### PR TITLE
chore: Revoke Access Tokens when a User is Deactivated

### PR DESCRIPTION
**Pull request by ShreyaLnuHpe**
## Ticket

[DET-10404](https://hpe-aiatscale.atlassian.net/browse/DET-10404)

## Checklist

- [ ] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[DET-10404]: https://hpe-aiatscale.atlassian.net/browse/DET-10404?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/user/postgres_users.go [hunk 1]
```diff
@@ -131,6 +131,13 @@ func Update(
 				Where("id = ?", updated.ID).Exec(ctx); err != nil {
 				return fmt.Errorf("error setting active status of %q: %s", updated.Username, err)
 			}
+			// Revoke all access tokens of a user when it is deactivated.
+			if !updated.Active {
+				err := revokeUserAccessTokens(ctx, tx, updated.ID)
```
Comment (corban-beaird) [inline master/internal/user/postgres_users.go:136]: what happens if the user being deactivated doesn't have any access tokens?

Comment (ShreyaLnuHpe) [reply master/internal/user/postgres_users.go:136]: The `Exec` function in Bun will still execute successfully, even if no rows are affected. The absence of matching rows is not considered an error by SQL standards, and Bun will not return an error for this case. Therefore, the function will return `nil`, indicating no errors occurred.

If you want to check whether any rows were affected by the update, you can inspect the result object returned by `Exec`, but IMO, having any log might create a noise in this case. 

```
result, err := tx.NewUpdate().
	Table("user_sessions").
	Set("revoked = ?", true).
	Where("user_id = ?", userID).
	Where("token_type = ?", model.TokenTypeAccessToken).
	Exec(ctx)
if err != nil {
	return err
}

rowsAffected, err := result.RowsAffected()
if err != nil {
	return err
}

if rowsAffected == 0 {
	// No rows were updated, handle accordingly
}
```

Comment (corban-beaird) [reply master/internal/user/postgres_users.go:136]: Perfect! 😄

**Review by corban-beaird — APPROVED** [review]
Fantastic work!

**Pull request merged**